### ISSUE TITLE
`FetchContent` usage results in a project that is always out-of-date

### ISSUE THREAD

**Issue by jeremyong** [cschreib/snitch#79]
After integrating the project via the CMake instructions (using CMake 3.24, Windows, Ninja generator), the project always reconfigures after each build. Unfortunately, I wasn't able to determine why the `generate.stamp` file is consistently out-of-date. Do you happen to know if this might be related to some aspect of the `snitch` CMake setup, or is this possibly an upstream bug? Thanks!

**Comment by cschreib** [comment]
Hello! I am using the exact code from the Readme to link to *snitch* in my personal projects, and never experienced this problem. However I exclusively compile these projects on linux with make, so there could be a problem with Microsoft-based toolchains.

I just tried compiling [one of these projects (observable_unique_ptr)](https://github.com/cschreib/observable_unique_ptr) on a Windows machine. With the default Visual Studio generator, no problem. I also tried with the Ninja generator, and still no problem. FYI, these are the commands I ran (from a Visual Studio Developer Command Prompt):

```cmd
cmake -G Ninja -DOUP_DO_TEST=1 -S . -B build
cmake --build build
```

From there on, running
```cmd
cmake --build build
```

... produces:
```cmd
ninja: no work to do.
```

Are you able to reproduce your problem when building [observable_unique_ptr](https://github.com/cschreib/observable_unique_ptr)?

NB:
 - Ninja version: 1.9.0
 - CMake version: 3.26.1
